src/components/UserProfile/UserProfile.jsx

```diff
@@ -645,16 +645,17 @@ function UserProfile(props) {
   const { userId: targetUserId } = props.match ? props.match.params : { userId: undefined };
   const { userid: requestorId, role: requestorRole } = props.auth.user;
 
-  /**  Login User's email */
-  const authEmail = props.auth?.user?.email;
+  const {role: userRole} = userProfile;
+  const authEmail = props.userProfile?.email;
```

Comment:
```suggestion
  const authEmail = props.auth?.user?.email;
```
See line 659. This has to be the email of auth. My understanding is that `userProfile.email` is the email of the account you're viewing.